vm/hv1/hv1_structs/src/proc_mask.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Structures for working with processor masks.
+
+/// A set of processor IDs, stored as a sparse array of 64-bit masks.
+#[derive(Clone)]
+pub struct ProcessorSet<'a> {
+    valid_masks: u64,
+    masks: &'a [u64],
+}
+
+impl<'a> ProcessorSet<'a> {
+    pub fn from_generic_set(format: u64, rest: &'a [u64]) -> Option<Self> {
+        if format != hvdef::hypercall::HV_GENERIC_SET_SPARSE_4K {
+            return None;
+        }
+        let &[valid_masks, ref masks @ ..] = rest else {
+            return None;
+        };
+        Self::from_processor_masks(valid_masks, masks)
+    }
+
+    pub fn from_processor_masks(valid_masks: u64, masks: &'a [u64]) -> Option<Self> {
+        let mask_count = valid_masks.count_ones();
+        if masks.len() != mask_count as usize {
+            return None;
+        }
+        Some(Self { valid_masks, masks })
+    }
+
+    /// Returns the set as a raw HV_GENERIC_SET_SPARSE_4K, suitable for use in a hypercall.
+    pub fn as_raw(&self) -> Vec<u64> {
```

Comment:
Doesn't seem like this is used anywhere.

But if it is, or if you want to keep it anyway, let's make it `write_raw` or something and have it write to a `&mut [u64]`, and document the required length.

Or just expose `valid_masks` and `masks`.

Reply:
Maybe the latter--I am thinking about how a caller would use this. They would probably need to write into an `&[AtomicU64]` or something.